### ISSUE TITLE
float16_t, float32_t,  float64_t, float128_t typedefs clashes 

### ISSUE THREAD

**Issue by ghost** [SDL-Hercules-390/hyperion#675]
the typedefs float16_t, float32_t,  float64_t, float128_t in softfloat_types.h clash 
with the corresponding typedefs in the system header arm_neon.h

this clash prevents the implementation of the arm neon intrinsics

the solution is pretty simple.... 
change floatxxx_t to something like softfloat_floatxxx_t/SFLT_floatxxx_t

with a smart editor the change can be done in couple of minutes
BTDTGTTS

done and working in my private repo

 posted also into the softfloat repository

adding into hstdinc.h

#include <arm_neon.h>

and trying to run gmake returned

```
  CC       ieee.lo
In file included from ../ieee.c:148:
In file included from /Users/enrico/hercules.SDL/work/extpkgs/include/softfloat.h:63:
/Users/enrico/hercules.SDL/work/extpkgs/include/softfloat_types.h:52:32: error: typedef redefinition with different types ('struct float16_t' vs '__fp16')
typedef struct { uint16_t v; } float16_t;
                               ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/arm_neon.h:40:16: note: previous definition is here
typedef __fp16 float16_t;
               ^
In file included from ../ieee.c:148:
In file included from /Users/enrico/hercules.SDL/work/extpkgs/include/softfloat.h:63:
/Users/enrico/hercules.SDL/work/extpkgs/include/softfloat_types.h:53:32: error: typedef redefinition with different types ('struct float32_t' vs 'float')
typedef struct { uint32_t v; } float32_t;
                               ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/arm_neon.h:39:15: note: previous definition is here
typedef float float32_t;
              ^
In file included from ../ieee.c:148:
In file included from /Users/enrico/hercules.SDL/work/extpkgs/include/softfloat.h:63:
/Users/enrico/hercules.SDL/work/extpkgs/include/softfloat_types.h:54:32: error: typedef redefinition with different types ('struct float64_t' vs 'double')
typedef struct { uint64_t v; } float64_t;
                               ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/arm_neon.h:42:16: note: previous definition is here
typedef double float64_t;
               ^
../ieee.c:394:18: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    int neg = (op.v & 0x8000000000000000ULL ) ? 1 : 0;
               ~~^~
../ieee.c:397:9: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    if (FLOAT64_ISNAN( op ))                {return float_class_pos_quiet_nan     >> neg;}
        ^~~~~~~~~~~~~~~~~~~
../ieee.c:257:39: note: expanded from macro 'FLOAT64_ISNAN'
#define FLOAT64_ISNAN( _op )    (((_op.v              & 0x7FF0000000000000ULL) == 0x7FF0000000000000ULL) &&  \
                                   ~~~^~
../ieee.c:397:9: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    if (FLOAT64_ISNAN( op ))                {return float_class_pos_quiet_nan     >> neg;}
        ^~~~~~~~~~~~~~~~~~~
../ieee.c:258:39: note: expanded from macro 'FLOAT64_ISNAN'
                                  (_op.v              & 0x000FFFFFFFFFFFFFULL) )
                                   ~~~^~
../ieee.c:398:13: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    if (!(op.v & 0x7FFFFFFFFFFFFFFFULL ))   {return float_class_pos_zero          >> neg;}
          ~~^~
../ieee.c:399:13: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    if ( (op.v & 0x7FFFFFFFFFFFFFFFULL )
          ~~^~
../ieee.c:401:13: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    if (  op.v & 0x7FF0000000000000ULL )    {return float_class_pos_normal        >> neg;}
          ~~^~
../ieee.c:407:18: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    int neg = (op.v & 0x80000000) ? 1 : 0;
               ~~^~
../ieee.c:410:9: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    if (FLOAT32_ISNAN( op ))        {return float_class_pos_quiet_nan     >> neg;}
        ^~~~~~~~~~~~~~~~~~~
../ieee.c:260:39: note: expanded from macro 'FLOAT32_ISNAN'
#define FLOAT32_ISNAN( _op )    (((_op.v              & 0x7F800000           ) == 0x7F800000           ) &&  \
                                   ~~~^~
../ieee.c:410:9: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    if (FLOAT32_ISNAN( op ))        {return float_class_pos_quiet_nan     >> neg;}
        ^~~~~~~~~~~~~~~~~~~
../ieee.c:261:39: note: expanded from macro 'FLOAT32_ISNAN'
                                  (_op.v              & 0x007FFFFF))
                                   ~~~^~
../ieee.c:411:13: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    if (!(op.v & 0x7FFFFFFF))       {return float_class_pos_zero          >> neg;}
          ~~^~
../ieee.c:412:13: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    if ( (op.v & 0x7FFFFFFF)
          ~~^~
../ieee.c:414:13: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    if (  op.v & 0x7F800000)        {return float_class_pos_normal        >> neg;}
          ~~^~
../ieee.c:591:7: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    op->v = *f;
    ~~^ ~
../ieee.c:596:12: error: member reference base type 'float64_t' (aka 'double') is not a structure or union
    *f = op->v;
         ~~^ ~
../ieee.c:601:7: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    op->v = *f;
    ~~^ ~
../ieee.c:606:12: error: member reference base type 'float32_t' (aka 'float') is not a structure or union
    *f = op->v;
         ~~^ ~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
gmake[2]: *** [Makefile:2637: ieee.lo] Error 1
gmake[1]: *** [Makefile:2687: all-recursive] Error 1
gmake: *** [Makefile:2006: all] Error 2
```

**Comment by Fish-Git** [comment]
Enrico, Bill,

Can you please apply the following patch and let me know whether it fixes your problem or not? If it does, I will go ahead and commit the fix. It is based on the following GitHub Issues comment made for another product that had the same problem: https://github.com/ucb-bar/berkeley-softfloat-3/issues/12#issuecomment-1534234747

* [fix675.zip](https://github.com/user-attachments/files/16366413/fix675.zip)

```diff
--- hyperion-1/ieee.c	2024-07-21 18:45:52.258647800 -0700
+++ hyperion-0/ieee.c	2024-07-24 11:43:06.264657000 -0700
@@ -145,6 +145,17 @@
   #define LITTLEENDIAN              /* Then #define LITTLEENDIAN macro       */
 #endif                              /* endif !defined( WORDS_BIGENDIAN )     */
 
+/* (arm_neon.h type conflict workaround) */
+#undef  float16_t
+#undef  float32_t
+#undef  float64_t
+#undef  float128_t
+
+#define float16_t   sfloat16_t
+#define float32_t   sfloat32_t
+#define float64_t   sfloat64_t
+#define float128_t  sfloat128_t
+
 #include "softfloat.h"              /* Master SoftFloat #include header      */
 
 /*****************************************************************************/
```

**Comment by Fish-Git** [comment]
Fixed by commit 61edad864dd6bfef54c62ce44b82b90a8acd81e1.

**Comment by wrljet** [comment]
Builds for me without trouble on macOS Sonoma 14.5, Apple M1 CPU, Clang 15.0.0.
